core/src/test/scala/za/co/absa/spline/harvester/postprocessing/DataSourceUriFilterSpec.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.harvester.postprocessing
+
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatestplus.mockito.MockitoSugar
+import za.co.absa.spline.harvester.HarvestingContext
+import za.co.absa.spline.producer.model.v1_1.WriteOperation
+
+class DataSourceUriFilterSpec extends AnyFlatSpec with Matchers with MockitoSugar {
+
+  it should "apply one filter" in {
+
+    def uri(pwd: String) = s"jdbc:sqlserver://database.windows.net:1433;user=sample;password=$pwd;" +
+      "encrypt=true;trustServerCertificate=false;hostNameInCertificate=*.database.windows.net;loginTimeout=30:"
+
+    val wop = WriteOperation(List.empty, uri("123456"), false, "42", List.empty, None, None)
+
+    val filter = new DataSourceUriFilter()
+    val pp = new PostProcessor(Seq(filter), mock[HarvestingContext])
+
+    val filteredOp = pp.process(wop)
+
+    filteredOp.outputSource shouldEqual uri("*****")
+  }
```

Comment:
Usually in unit-testing, DAMP principle should be preferred over DRY. 
Trivial constant and code repetition leads to more readable and descriptive code, which considering unit-tests purposes is beneficial.
Consider this variant. It's longer but arguably more readable
```suggestion
  it should "apply one filter" in {
    val wop = WriteOperation(
      Nil,
      "" +
        "jdbc:sqlserver://database.windows.net:1433" +
        ";user=sample" +
        ";password=123456" +
        ";encrypt=true" +
        ";trustServerCertificate=false" +
        ";hostNameInCertificate=*.database.windows.net" +
        ";loginTimeout=30:",
      append = false, "", Nil, None, None)

    val filter = new DataSourceUriFilter()
    val pp = new PostProcessor(Seq(filter), mock[HarvestingContext])
    val filteredOp = pp.process(wop)

    filteredOp.outputSource shouldEqual "" +
      "jdbc:sqlserver://database.windows.net:1433" +
      ";user=sample" +
      ";password=*****" + // <-- PASSWORD SHOULD BE SANITIZED
      ";encrypt=true" +
      ";trustServerCertificate=false" +
      ";hostNameInCertificate=*.database.windows.net" +
      ";loginTimeout=30:"
  }
```

Reply:
Also notice that that test method tests two classes at once: `DataSourceUriFilter` and `PostProcessor`. 
A unit test should (almost) always only test _one_ class and _one_ method at a time.
(Perhaps the only exception from this rule could be a situation when you test a pair of mutually related classes, like read/write, get/set etc, testing which in complete isolation is often difficult)